LVGL sample cannot be built with CONFIG_LEGACY_INCLUDE_PATH=n

**Describe the bug**
Building the `zephyr/samples/subsys/display/lvgl/` sample throws an error if being built with non-legacy include paths (i.e. `CONFIG_LEGACY_INCLUDE_PATH=n` Kconfig setting). Both legacy and non-legacy build paths should be supported on Zephyr v3.1.0.

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using? Any target with `CONFIG_LEGACY_INCLUDE_PATH=n`
 - What have you tried to diagnose or workaround this issue? Changing the `LV_TICK_CUSTOM_INCLUDE` define in `modules/lib/gui/lvgl/zephyr/lv_conf.h` file from `"kernel.h"` to `<zephyr/kernel.h>` fixes the issue for non-legacy include paths mode.

**To Reproduce**
Steps to reproduce the behavior:
1. Append line `CONFIG_LEGACY_INCLUDE_PATH=n` to `zephyr/samples/subsys/display/lvgl/prj.conf` file.
2. While in sample directory, execute `west build -p -b stm32h7b3i_dk` (or any other board with display support)
3. See error

**Expected behavior**
Sample should be buildable with both legacy and non-legacy include path modes.

**Impact**
Inconvenience

**Logs and console output**
```
[156/357] Building C object modules/lvgl/CMakeFiles/..__m...hyrproject/modules/lib/gui/lvgl/src/hal/lv_hal_tick.c.obj
FAILED: modules/lvgl/CMakeFiles/..__modules__lib__gui__lvgl__zephyr.dir/home/users/tmilkovic/zephyrproject/modules/lib/gui/lvgl/src/hal/lv_hal_tick.c.obj 
ccache /home/users/tmilkovic/.local/opt/zephyr-sdk-0.14.2/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DCORE_CM7 -DHSE_VALUE=24000000 -DKERNEL -DLV_CONF_INCLUDE_SIMPLE=1 -DLV_CONF_PATH=../zephyr/lv_conf.h -DSTM32H7B3xxQ -DUSE_FULL_LL_DRIVER -DUSE_HAL_DRIVER -D_FORTIFY_SOURCE=2 -D__PROGRAM_START -D__ZEPHYR__=1 -I/home/users/tmilkovic/zephyrproject/zephyr/lib/gui/lvgl -I/home/users/tmilkovic/zephyrproject/zephyr/include -I/home/users/tmilkovic/zephyrproject/zephyr/samples/subsys/display/lvgl/build/zephyr/include/generated -I/home/users/tmilkovic/zephyrproject/zephyr/soc/arm/st_stm32/stm32h7 -I/home/users/tmilkovic/zephyrproject/zephyr/drivers -I/home/users/tmilkovic/zephyrproject/zephyr/soc/arm/st_stm32/common -I/home/users/tmilkovic/zephyrproject/modules/hal/cmsis/CMSIS/Core/Include -I/home/users/tmilkovic/zephyrproject/modules/hal/atmel/include -I/home/users/tmilkovic/zephyrproject/modules/hal/stm32/stm32cube/stm32h7xx/soc -I/home/users/tmilkovic/zephyrproject/modules/hal/stm32/stm32cube/stm32h7xx/drivers/include -I/home/users/tmilkovic/zephyrproject/modules/hal/stm32/stm32cube/stm32h7xx/drivers/include/Legacy -I/home/users/tmilkovic/zephyrproject/modules/hal/stm32/stm32cube/common_ll/include -I/home/users/tmilkovic/zephyrproject/modules/lib/gui/lvgl/zephyr/../src -isystem /home/users/tmilkovic/zephyrproject/zephyr/lib/libc/minimal/include -isystem /home/users/tmilkovic/.local/opt/zephyr-sdk-0.14.2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.3.0/include -isystem /home/users/tmilkovic/.local/opt/zephyr-sdk-0.14.2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.3.0/include-fixed -Os -imacros /home/users/tmilkovic/zephyrproject/zephyr/samples/subsys/display/lvgl/build/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-m7 -mthumb -mabi=aapcs -mfp16-format=ieee -imacros /home/users/tmilkovic/zephyrproject/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-reorder-functions -fno-defer-pop -fmacro-prefix-map=/home/users/tmilkovic/zephyrproject/zephyr/samples/subsys/display/lvgl=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/users/tmilkovic/zephyrproject/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/users/tmilkovic/zephyrproject=WEST_TOPDIR -ffunction-sections -fdata-sections -std=c99 -nostdinc -MD -MT modules/lvgl/CMakeFiles/..__modules__lib__gui__lvgl__zephyr.dir/home/users/tmilkovic/zephyrproject/modules/lib/gui/lvgl/src/hal/lv_hal_tick.c.obj -MF modules/lvgl/CMakeFiles/..__modules__lib__gui__lvgl__zephyr.dir/home/users/tmilkovic/zephyrproject/modules/lib/gui/lvgl/src/hal/lv_hal_tick.c.obj.d -o modules/lvgl/CMakeFiles/..__modules__lib__gui__lvgl__zephyr.dir/home/users/tmilkovic/zephyrproject/modules/lib/gui/lvgl/src/hal/lv_hal_tick.c.obj -c /home/users/tmilkovic/zephyrproject/modules/lib/gui/lvgl/src/hal/lv_hal_tick.c
In file included from /home/users/tmilkovic/zephyrproject/modules/lib/gui/lvgl/src/hal/../lv_conf_internal.h:35,
                 from /home/users/tmilkovic/zephyrproject/modules/lib/gui/lvgl/src/hal/lv_hal_tick.h:16,
                 from /home/users/tmilkovic/zephyrproject/modules/lib/gui/lvgl/src/hal/lv_hal_tick.c:9:
/home/users/tmilkovic/zephyrproject/modules/lib/gui/lvgl/src/hal/../../zephyr/lv_conf.h:34:33: fatal error: kernel.h: No such file or directory
   34 | #define LV_TICK_CUSTOM_INCLUDE  "kernel.h"
      |                                 ^~~~~~~~~~
compilation terminated.
[165/357] Building C object modules/lvgl/CMakeFiles/..__m...rproject/modules/lib/gui/lvgl/src/widgets/lv_slider.c.obj
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /usr/bin/cmake --build /home/users/tmilkovic/zephyrproject/zephyr/samples/subsys/display/lvgl/build
```

**Environment (please complete the following information):**
 - OS: Debian 12 - Bookworm (testing)
 - Toolchain Zephyr SDK v0.14.2
 - Zephyr commit SHA: 4cc5c56d8c4a4727bcfb6a49cdc2d6ffba39295f

**Additional context**
LVGL module needs to be modified in order to fix this. File `zephyr/lv_conf.h` needs to be changed from:

```
#define LV_TICK_CUSTOM 1
#define LV_TICK_CUSTOM_INCLUDE "kernel.h"
#define LV_TICK_CUSTOM_SYS_TIME_EXPR (k_uptime_get_32())
```

to something like:

```
#define LV_TICK_CUSTOM 1
#ifdef CONFIG_LEGACY_INCLUDE_PATH
#define LV_TICK_CUSTOM_INCLUDE "kernel.h"
else
#define LV_TICK_CUSTOM_INCLUDE <zephyr/kernel.h>
#endif /* CONFIG_LEGACY_INCLUDE_PATH */
#define LV_TICK_CUSTOM_SYS_TIME_EXPR (k_uptime_get_32())
```

--- a/west.yml
+++ b/west.yml
@@ -163,7 +163,7 @@ manifest:
       revision: 12019623bbad9eb54fe51066847a7cbd4b4eac57
       path: modules/lib/loramac-node
     - name: lvgl
-      revision: 9955c3e1e4f361884e938dc016da5a715a4527fe
+      revision: df717ac87a9fd80246cc8df24554475e59bda21f
       path: modules/lib/gui/lvgl
     - name: lz4
       revision: 8e303c264fc21c2116dc612658003a22e933124d

